### PR TITLE
[4.0] Remove deprecated cache handlers, cleanup cache/session config related to deprecations

### PR DESCRIPTION
**Pull request by mbabker**
Pull Request for Issue # .

### Summary of Changes

- Removes the deprecated cache handlers as marked in https://github.com/joomla/joomla-cms/pull/19112
- Removes associated CI configurations for these handlers
- Removes configuration related to these deprecated handlers
- Removes configuration related to session handlers which have been removed (basically the same things but session instead of cache "namespace")
- Removes language strings for removed cache/session handlers

### Testing Instructions

- Code review, CMS works as long as you don't have any of these deprecated things set up in your configuration

### Documentation Changes Required

- Document removals

### CODE REVIEW

File: administrator/components/com_admin/Model/SysinfoModel.php [hunk 1]
```diff
@@ -115,8 +115,6 @@ class SysInfoModel extends BaseDatabaseModel
 			'live_site',
 			'log_path',
 			'mailfrom',
-			'memcache_server_host',
-			'memcached_server_host',
```
Comment (Quy) [inline administrator/components/com_admin/Model/SysinfoModel.php]: Revert line 119?

Comment (mbabker) [reply administrator/components/com_admin/Model/SysinfoModel.php]: Oh I'm so glad to get rid of one of these two extremely closely named handlers.

File: administrator/language/en-GB/en-GB.lib_joomla.ini [hunk 2]
```diff
@@ -328,24 +328,16 @@ JLIB_FORM_VALIDATE_FIELD_INVALID="Invalid field: %s"
...
 JLIB_FORM_VALUE_CACHE_REDIS="Redis"
 JLIB_FORM_VALUE_CACHE_WINCACHE="Windows Cache"
-JLIB_FORM_VALUE_CACHE_XCACHE="XCache"
-JLIB_FORM_VALUE_SESSION_APC="Alternative PHP Cache"
 JLIB_FORM_VALUE_SESSION_APCU="APC User Cache"
 JLIB_FORM_VALUE_SESSION_DATABASE="Database"
 JLIB_FORM_VALUE_SESSION_FILESYSTEM="Filesystem"
-JLIB_FORM_VALUE_SESSION_MEMCACHE="Memcache"
 JLIB_FORM_VALUE_SESSION_MEMCACHED="Memcached (Experimental)"
-JLIB_FORM_VALUE_SESSION_REDIS="Redis"
```
Comment (csthomas) [inline administrator/language/en-GB/en-GB.lib_joomla.ini]: IMO Redis should stay.

**Comment by mbabker** [comment]
Ping?

Also, all the tests are gonna be broken until whatever repo is updated to remove the test cases in cache that cover these removed classes (this PR did that once upon a time but now that you have to do 3 or 4 PRs to patch production code and tests...).

**Review by brianteeman — APPROVED** [review]
Approve language changes

**Review by laoneo — APPROVED** [review]

**Pull request merged**